arch/x86/crc32_pclmulqdq_tpl.h

```diff
@@ -246,18 +245,27 @@ static inline void crc32_fold_save(__m128i *fold, const __m128i *fold0, const __
     _mm_storeu_si128(fold + 3, *fold3);
 }
 
-Z_INTERNAL uint32_t crc32_fold_pclmulqdq_reset(crc32_fold *crc) {
+Z_INTERNAL uint32_t CRC32_FOLD_RESET(crc32_fold *crc) {
     __m128i xmm_crc0 = _mm_cvtsi32_si128(0x9db42487);
     __m128i xmm_zero = _mm_setzero_si128();
     crc32_fold_save((__m128i *)crc->fold, &xmm_crc0, &xmm_zero, &xmm_zero, &xmm_zero);
     return 0;
 }
 
-#define ONCE(op)            if (first) { first = 0; op; }
-#define XOR_INITIAL(where)  ONCE(where = _mm_xor_si128(where, xmm_initial))
+#define ONCE(op)               if (first) { first = 0; op; }
+#define XOR_INITIAL128(where)  ONCE(where = _mm_xor_si128(where, xmm_initial))
+#ifdef X86_VPCLMULQDQ
+#  define XOR_INITIAL512(where)  ONCE(where = _mm512_xor_si512(where, zmm_initial))
+#endif
 
+#ifdef X86_VPCLMULQDQ
+#  include "crc32_fold_vpclmulqdq_tpl.h"
+#endif
 #include "crc32_fold_pclmulqdq_tpl.h"
```

Comment:
Perhaps pcl should be included above include for vpcl?
```c
#include "crc32_fold_pclmulqdq_tpl.h"
#ifdef X86_VPCLMULQDQ
#  include "crc32_fold_vpclmulqdq_tpl.h"
#endif
```

Reply:
The file ``crc32_fold_vpclmulqdq_tpl.h`` defines the functions that are used in ``crc32_fold_pclmulqdq_tpl.h``.
Therefore, I placed them in that order.

The order can be changed, since the prototypes of these functions are defined above.

Reply:
I didn't know if there was a reason or not. Either way is ok.